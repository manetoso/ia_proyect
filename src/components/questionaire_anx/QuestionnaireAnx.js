import React, { useState } from "react";

export default function QuestionnaireAnx(props) {
  const [optionSelected, setOptionSelected] = useState();

  const handleChooise = (e) => {
    // console.log(e.target.value);
    setOptionSelected(e.target.value);
    props.action(props.number, e.target.value);
  };

  return (
    <div className="card border-secondary mb-3">
      <div className="card-header">Pregunta {props.number}</div>
      <div className="card-body">
        <p className="card-text">{props.question}</p>
        <div
          className="form-group"
          style={{ display: "flex", justifyContent: "space-around" }}
          onChange={handleChooise}
        >
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="radio"
                className="form-check-input"
                value="1"
                checked={optionSelected === "1"}
              />
              <span>1</span>
            </label>
          </div>
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="radio"
                className="form-check-input"
                value="2"
                checked={optionSelected === "2"}
              />
              <span>2</span>
            </label>
          </div>
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="radio"
                className="form-check-input"
                value="3"
                checked={optionSelected === "3"}
              />
              <span>3</span>
            </label>
          </div>
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="radio"
                className="form-check-input"
                value="4"
                checked={optionSelected === "4"}
              />
              <span>4</span>
            </label>
          </div>
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="radio"
                className="form-check-input"
                value="5"
                checked={optionSelected === "5"}
              />
              <span>5</span>
            </label>
          </div>
        </div>
      </div>
    </div>
  );
}
