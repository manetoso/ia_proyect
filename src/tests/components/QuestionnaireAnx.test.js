import React from "react";
import { mount, shallow } from "enzyme";

import QuestionnaireAnx from "../../components/questionaire_anx/QuestionnaireAnx";
import { act, fireEvent } from "@testing-library/react";

//LINK DE LA DOCUMENTACIÓN EN ENZYME CON SHALLOW https://enzymejs.github.io/enzyme/docs/api/shallow.html

//SE CREA EL DOCUMENTO DE PRUEBAS
describe("Testing in <QuestionnaireAnx />", () => {
  //VARIABLES QUE LLENAN LOS CAMPOS NECESARIOS PARA REALIZAR PRUEBAS EN EL COMPONENTE
  const number = 1;
  const question = "¿Está usted bien?";

  //INTICA QUE EL WRAPPER ES EL CONTENIDO HTML DEL COMPONENTE
  const wrapper = shallow(
    <QuestionnaireAnx action={() => {}} number={number} question={question} />
  );

  //TODOS LOS TESTS SON PRUEBAS UNITARIAS DENTRO DE ESTE DOCUMENTO DE PRUEBAS

  //TEST QUE BUSCA QUE SE MUESTRE CORRECTAMENTE EL CONTENIDO HTML DEL COMPONENTE
  test("should show the component correctly", () => {
    //ESCRIBE UN NUEVO SNAPTSHOT EN LA CARPETA DE _snapshots_
    //DENTRO DEL DIRECTORIO DE PRUEBAS DEL COMPONENTE AL EJECUTARCE POR PRIMERA VEZ

    //SI EXISTE UNA SNAPTSHOT LA USA PARA COMPARAR EL WRAPPER CON ESTA
    //SI SON IGUALES TODO ESTÁ BIEN, SI NO SE PARECEN EN ALGO LA PRUEBA FALLA
    expect(wrapper).toMatchSnapshot();
  });

  //TEST QUE BUSCA QUE EL NÚMERO DE PREGUNTA ESTÉ DENTRO DE UN <div/>
  //CON EL CLASSNAME DE "card-header"
  test('should show the question number on a div with the className "card-header"', () => {
    const questionNumberTest = wrapper.find("div.card-header");
    const textNumber = "Pregunta " + number;

    expect(questionNumberTest.hasClass("card-header")).toBe(true);
    expect(questionNumberTest.text().trim()).toBe(textNumber);
  });

  //TEST QUE BUSCA QUE LA PREGUNTA ESTÉ DENTRO DE UN <p/>
  //CON EL CLASSNAME "card-text"
  test('should show the question on a paragraph with the className "card-text"', () => {
    const questionTest = wrapper.find("p.card-text");

    expect(questionTest.hasClass("card-text")).toBe(true);
    expect(questionTest.text().trim()).toBe(question);
  });

  //TEST QUE BUSCA QUE SE MUESTREN 5 POSIBLES RESPUESTAS A ESCOGER
  test("should give 5 answers to choose", () => {
    const answersLength = wrapper.find("div.form-group").children();

    expect(answersLength).toHaveLength(5);
  });

  test("check if the user can change their anwers before submit tests", () => {
    //
  });
});
