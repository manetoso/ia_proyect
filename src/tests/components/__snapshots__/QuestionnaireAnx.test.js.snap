// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <QuestionnaireAnx /> should show the component correctly 1`] = `
<div
  className="card border-secondary mb-3"
>
  <div
    className="card-header"
  >
    Pregunta 
    1
  </div>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      ¿Está usted bien?
    </p>
    <div
      className="form-group"
      onChange={[Function]}
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-around",
        }
      }
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            type="radio"
            value="0"
          />
          <span>
            1
          </span>
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            type="radio"
            value="2"
          />
          <span>
            2
          </span>
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            type="radio"
            value="3"
          />
          <span>
            3
          </span>
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            type="radio"
            value="4"
          />
          <span>
            4
          </span>
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            type="radio"
            value="5"
          />
          <span>
            5
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
